### PR TITLE
feat: Check that makeInstance() returns an actual invite

### LINKED ISSUE #820: [Zoe] Check that invite field of `makeInstance` return value is an actual invite
No check is performed on the `invite` field ; it is returned as it is

https://github.com/Agoric/agoric-sdk/blob/7ceebe5f45333525eb59a9d6ec7a062f93ea2da7/packages/zoe/src/zoe.js#L336-L342

Edit below by Kate 4/7:
We should add a check:

```inviteIssuer.isLive(invite)```
This will take another turn since isLive returns a promise.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
Includes a test.

closes #820

### CODE REVIEW

File: packages/zoe/src/zoe.js [hunk 1]
```diff
@@ -543,7 +543,12 @@ const makeZoe = (additionalEndowments = {}) => {
               // Once the contract is made, we add the publicAPI to the
               // contractRecord
               instanceTable.update(instanceHandle, { publicAPI });
-              return invite;
+              return inviteIssuer.isLive(invite).then(success => {
+                if (success) {
```
Comment (katelynsills) [inline packages/zoe/src/zoe.js]: Can we use an assert here and reverse the return and the error check?
```suggestion
assert(success, details`invites must be issued by the inviteIssuer.`);
return invite;
```

File: packages/zoe/test/unitTests/contracts/brokenAutoRefund.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        return `The offer was accepted`;
+      },
+    });
+    const { invite, inviteHandle } = zoe.makeInvite(seat, {
+      seatDesc: 'getRefund',
+    });
+    return invite;
+  };
+  return harden({
+    // should be makeSeatInvite(). Intentionally wrong to provoke an error.
```
Comment (katelynsills) [inline packages/zoe/test/unitTests/contracts/brokenAutoRefund.js:23]: Thanks for this comment. The comment made it easy to understand

File: packages/zoe/test/unitTests/contracts/test-brokenContract.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const { moolaR } = setup();
+  const zoe = makeZoe({ require });
+  // Pack the contract.
+  const { source, moduleFormat } = await bundleSource(automaticRefundRoot);
+  const installationHandle = zoe.install(source, moduleFormat);
+
+  const issuerKeywordRecord = harden({ Contribution: moolaR.issuer });
+
+  // 1: Alice tries to create an instance, but the contract is badly written.
+  zoe.makeInstance(installationHandle, issuerKeywordRecord).then(
```
Comment (katelynsills) [inline packages/zoe/test/unitTests/contracts/test-brokenContract.js]: I think we should use `t.throws` here. Is there a reason not to?

**Review by katelynsills — CHANGES_REQUESTED** [review]
Feel free to let me know if the changes I suggested aren't desirable.

File: packages/zoe/test/unitTests/contracts/test-brokenContract.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/zoe/test/unitTests/contracts/test-brokenContract.js]: `makeInstance` is async. I couldn't find a way to `await` the result.  It doesn't throw until the test is done.

File: packages/zoe/src/zoe.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/zoe/src/zoe.js]: done.

File: packages/zoe/test/unitTests/contracts/test-brokenContract.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/zoe/test/unitTests/contracts/test-brokenContract.js]: t.rejects() works.

File: packages/zoe/test/unitTests/contracts/test-brokenContract.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import harden from '@agoric/harden';
+
+import { makeZoe } from '../../../src/zoe';
+// TODO: Remove setupBasicMints and rename setupBasicMints2
+import { setup } from '../setupBasicMints2';
+
+const automaticRefundRoot = `${__dirname}/brokenAutoRefund`;
+
+test.only('zoe - brokenAutomaticRefund', async t => {
```
Comment (katelynsills) [inline packages/zoe/test/unitTests/contracts/test-brokenContract.js]: Please remove test.only

**Review by katelynsills — APPROVED** [review]
Let's remove the test.only and then it looks good to me!

File: packages/zoe/test/unitTests/contracts/test-brokenContract.js (hunk 4 above)
Comment (Chris-Hibbert) [reply packages/zoe/test/unitTests/contracts/test-brokenContract.js]: done.

**Pull request merged**